molo/core/content_import/views.py

```diff
@@ -7,13 +7,28 @@
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.decorators import parser_classes
 
-from molo.core.content_import import api
 from unicore.content.models import Localisation
 
+from molo.core.content_import import api
+from molo.core.content_import.errors import SiteResponseError
+
 
 @api_view(['GET'])
-def get_repos(request):
-    return Response({'repos': api.get_repo_summaries()})
+def get_repo_summaries(request):
+    params = request.query_params
```

Comment:
Yes, the client side normalises the input url and parses it into url components we give to the API.

There's a few reasons why I went with this approach:
  - parsing user input is the client-side's job, parsing data that machines understand is the api's job
  - we're restricting the api so that clients can't provide invalid urls (restricting an interface to prevent errors is better than telling the interface client that they gave erroneous input)
  - similar to the previous point, we are restricting the api so that clients can't provide urls that are valid, but that we can't use (e.g. `www.foo.bar/?baz=quux#corge`).
  - it allows us to to later change the UI to tell the user if they have entered invalid urls, or urls that we can't use